torchgeo/datasets/benin_cashews.py

```diff
@@ -17,7 +17,7 @@
 from torch import Tensor
 
 from .geo import NonGeoDataset
-from .utils import check_integrity, download_radiant_mlhub_dataset, extract_archive
+from .utils import check_integrity, download_radiant_mlhub_collection, extract_archive
```

Comment:
Let's remove the `download_radiant_mlhub_dataset` method, it's broken for newer versions of radiant-mlhub

Reply:
So `download_radiant_mlhub_dataset` is still required by SpaceNet6 as its `Collection` interface is different. That one still works as is now but it does not have md5 checks currently.